Error while running tests on OS X

I keep getting these failures during the basher test run on OS X.  There isn't a lot of info about what the actual error is, what code is being run, etc.

```
 ✓ extracts only usage
 ✓ multiline usage section
 /var/folders/vv/yvb4m22n5f511tbzjjjwcm240000gn/T/bats.44779.src: line 66: syntax error near unexpected token `>'                                                                                                                    51/119
 ✓ multiline extended help section

119 tests, 0 failures

make: *** [test] Error 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,8 +1,39 @@
-install: git clone https://github.com/sstephenson/bats.git
-before_script:
-  - git config --global user.email "user@example.com"
-  - git config --global user.name "User Name"
-script: bats/bin/bats --tap tests
-# skips unnecessary Ruby-specific setup
-language: c
-sudo: false
+language: bash
+
+os:
+- linux
+
+env:
+  - BASHVER=
+  - BASHVER=3.2
+  - BASHVER=4.0
+  - BASHVER=4.1
+  - BASHVER=4.2
+  - BASHVER=4.3
+  - BASHVER=4.4
+
+matrix:
+  include:
+    - os: osx
+
+services:
+  - docker
+
+script:
+- |
+  wget --no-check-certificate -O bats.tar.gz https://github.com/bats-core/bats-core/archive/v1.1.0.tar.gz
+  mkdir bats
+  tar --strip-components=1 -xvzf bats.tar.gz -C bats
+  if [[ "$TRAVIS_OS_NAME" == 'linux' && -n "$BASHVER" ]]; then
+    docker build --build-arg bashver=${BASHVER} --tag basher/basher:bash-${BASHVER} .
+    docker run -it bash:${BASHVER} --version
+    time docker run -it basher/basher:bash-${BASHVER} --tap /opt/basher/tests
+  else
+    git config --global user.email "user@example.com"
+    git config --global user.name "User Name"
+    time bats/bin/bats --tap tests
+  fi
+
+notifications:
+  email:
+    on_success: never

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,11 @@
+ARG bashver=latest
+
+FROM bash:${bashver}
+
+RUN apk add --no-cache git; \
+git config --global user.email "user@example.com"; \
+git config --global user.name "User Name";
+
+COPY . /opt/basher/
+
+ENTRYPOINT ["bash", "/opt/basher/bats/bin/bats"]

--- a/tests/basher-init.bats
+++ b/tests/basher-init.bats
@@ -63,7 +63,7 @@ load test_helper
 }
 
 hasShell() {
-  which "$1" &>>/dev/null
+  which "$1" >>/dev/null 2>&1
 }
 
 @test "is sh-compatible" {

